oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/oidc/authentication/OidcAuthorizationCodeAuthenticationProvider.java

```diff
@@ -154,6 +162,19 @@ public Authentication authenticate(Authentication authentication) throws Authent
 		}
 		OidcIdToken idToken = createOidcToken(clientRegistration, accessTokenResponse);
 
+		String nonceHash;
+		try {
+			nonceHash = createHash(authorizationRequest.getAttribute(IdTokenClaimNames.NONCE));
+		} catch (NoSuchAlgorithmException e) {
+			OAuth2Error nonceVerificationError = new OAuth2Error(UNVERIFIABLE_NONCE, e.getMessage(), null);
+			throw new OAuth2AuthenticationException(nonceVerificationError);
+		}
+		String nonceClaim = idToken.getClaim(IdTokenClaimNames.NONCE);
+		if (nonceClaim == null || !nonceClaim.equals(nonceHash)) {
```

Comment:
The `nonceClaim` is actually the `nonce` hash so let's rename to `nonceHashClaim`